frontend/src/components/common/protected-route/protected-route.tsx

```diff
@@ -21,6 +24,19 @@ const ProtectedRoute: FC<Props> = ({
     return <Navigate to={redirectTo} />;
   }
 
+  if (pagePermissions) {
+    const userPermissions = user?.permissions.map((item) => item.key) ?? [];
```

Comment:
Is there a need for `?` in `user?.permissions` ? Because there's a check upper
```
if (!hasUser) {
    return <Navigate to={redirectTo} />;
}
```